samples/nrf5340/netboot/src/main.c

```diff
@@ -39,10 +42,26 @@ int main(void)
 
 	uint32_t s0_addr = s0_address_read();
 	bool valid = false;
-	uint8_t status = pcd_fw_copy_status_get();
+
+	switch (pcd_fw_copy_status_get()) {
+#ifdef CONFIG_PCD_LOCK_NETCORE_DEBUG
+	case PCD_STATUS_LOCK_DEBUG:
+		nrfx_nvmc_word_write((uint32_t)&NRF_UICR_NS->APPROTECT,
```

Comment:
This should check it is not set before writing it, otherwise you're pointlessly writing over and over to the same region every time it boots

Reply:
This is not run every time the device boots. The reason for this is that the PCD_STATUS_LOCK_DEBUG is only set when we are transitioning from provisioning LCS to secure LCS, in which we then proceed to lock the UICR.APPROTECT.

Furthermore, the UICR.APPROTECT is writable only once, which I think is implemented in a way that you can only transform bits from 1 to 0. We are writing all zeroes in here. That said, you could actually write zeroes multiple times to this register. I don't think that it would increase the wear on the flash (even if it goes that far), as there is no change on the value.